src/devices/RadioReceiver/Devices/Tea5767/Tea5767.cs

```diff
@@ -0,0 +1,259 @@
+        private FrequencyRange GetFrequencyRange()
+        {
+            return (FrequencyRange)((_registers[3] & 0b_1101_1111) >> 5);
+        }
+
+        /// <summary>
+        /// Set TEA5767 FM frequency.
+        /// </summary>
+        /// <param name="frequency">FM frequency.</param>
+        private void SetFrequency(double frequency)
+        {
+            switch (FrequencyRange)
+            {
+                case FrequencyRange.Japan:
+                    if (frequency < 76 || frequency > 90)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(frequency), $"{nameof(frequency)} needs to be in the range of 76 to 90.");
+                    }
+                    break;
+                case FrequencyRange.Other:
+                    if (frequency < 87 || frequency > 108)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(frequency), $"{nameof(frequency)} needs to be in the range of 87 to 108.");
+                    }
+                    break;
+                default:
+                    break;
+            }
+
+            int f = (int)((frequency * 1000000 + 225000) / 8192);
+
+            byte high = (byte)((f & 0b_0011_1111_0000_0000) >> 8);
+            byte low = (byte)(f & 0b_1111_1111);
+
+            _registers[0] &= 0b_1100_0000;
+            _registers[0] |= high;
+            _registers[1] = low;
+
+            SaveRegisters();
+        }
+
+        /// <summary>
+        /// Get TEA5767 FM frequency.
+        /// </summary>
+        /// <returns>FM frequency.</returns>
+        private double GetFrequency()
+        {
+            byte[] readBuffer = ReadRegisters();
+
+            int f = ((readBuffer[0] & 0b_0011_1111) << 8) | readBuffer[1];
+
+            return Math.Round((f * 8192 - 225000) / 1000000.0, 1);
+        }
+
+        /// <summary>
+        /// Set whether TEA5767 is standby.
+        /// </summary>
+        /// <param name="isStandby">Standby if the value is true.</param>
+        private void SetStandby(bool isStandby)
+        {
+            if (isStandby)
+            {
+                _registers[3] |= 0b_0100_0000;
+            }
+            else
+            {
+                _registers[3] &= 0b_1011_1111;
+            }
+
+            SaveRegisters();
+        }
+
+        /// <summary>
+        /// Get whether TEA5767 is standby.
+        /// </summary>
+        /// <returns>Standby if the value is true.</returns>
+        private bool GetStandby()
+        {
+            return (_registers[3] & 0b_0100_0000) > 0 ? true : false;
+        }
+
+        /// <summary>
+        /// Automatic search for effective radio.
+        /// </summary>
+        /// <param name="isUp">If true, search up from the current frequency.</param>
+        /// <param name="stopLevel">Stop search condition (range from 1 to 3).</param>
+        public void Search(bool isUp = true, int stopLevel = 3)
```

Comment:
I'm slightly confused by `isUp` name. If this is false will it search downward? if so perhaps rename arg to `searchDirection` and type to something like:

```csharp
enum SearchDirection
{
    Up,
    Down,
}
```